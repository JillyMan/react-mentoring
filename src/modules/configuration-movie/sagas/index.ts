import {
    deleteMovieSaga,
    loadMovieSaga,
    saveNewMovieSaga,
    updateMovieSaga,
} from './configuration-movie-saga';

export const configurationMovieSagas = [
    saveNewMovieSaga,
    updateMovieSaga,
    deleteMovieSaga,
    loadMovieSaga,
];
