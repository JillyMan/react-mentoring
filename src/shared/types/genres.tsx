export const configurationGenres = ['Action', 'Documentary', 'Comedy', 'Horror', 'Crime'];

export const allGenres = ['All', ...configurationGenres];
