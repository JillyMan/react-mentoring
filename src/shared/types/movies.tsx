export interface MovieConfig {
    id: number;
    title: string;
    tagline: string;
    vote_average: number;
    vote_count: number;
    release_date: string;
    poster_path: string;
    overview: string;
    budget: number;
    revenue: number;
    genres: string[];
    runtime: number;
}

export const initialMovieConfig: MovieConfig = {
    id: 0,
    title: '',
    tagline: '',
    vote_average: 0,
    vote_count: 0,
    release_date: '',
    poster_path: '',
    overview: '',
    budget: 0,
    revenue: 0,
    genres: [],
    runtime: 0,
};

export const movieConfigNames = {
    id: 'id',
    title: 'title',
    tagline: 'tagline',
    vote_average: 'vote_average',
    vote_count: 'vote_count',
    release_date: 'release_date',
    poster_path: 'poster_path',
    overview: 'overview',
    budget: 'budget',
    revenue: 'revenue',
    genres: 'genres',
    runtime: 'runtime',
};
