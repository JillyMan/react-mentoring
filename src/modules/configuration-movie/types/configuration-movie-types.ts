import { MovieConfig, movieConfigNames } from 'shared/types/movies';

export const requiredFildsName = [
    movieConfigNames.title,
    movieConfigNames.release_date,
    movieConfigNames.poster_path,
    movieConfigNames.vote_average,
    movieConfigNames.genres,
    movieConfigNames.runtime,
    movieConfigNames.overview,
];
