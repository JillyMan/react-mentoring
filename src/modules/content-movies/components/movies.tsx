import React from 'react';
import { Grid } from '@mui/material';
import { MovieCardWithError } from './movie-card';
import { MovieConfig } from 'shared/types/movies';

interface Props {
    movies: MovieConfig[];
    onMovieClick: (movie: MovieConfig) => void;
    onDeleteMovie: (id: number) => void;
    onUpdateMovie: (movei: MovieConfig) => void;
}

export const Movies = ({ movies, onMovieClick, onDeleteMovie, onUpdateMovie }: Props) => {
    return (
        <Grid container rowSpacing={3} spacing={3}>
            {movies.map((movie, id) => (
                <Grid item xs={3} key={movie.id}>
                    <MovieCardWithError
                        onMovieClick={onMovieClick}
                        onDeleteMovie={onDeleteMovie}
                        onUpdateMovie={onUpdateMovie}
                        movie={movie}
                    />
                </Grid>
            ))}
        </Grid>
    );
};
