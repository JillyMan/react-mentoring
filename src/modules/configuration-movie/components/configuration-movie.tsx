import React from 'react';
import DatePicker from '@mui/lab/DatePicker';
import SendIcon from '@mui/icons-material/Send';
import {
    Grid,
    Typography,
    InputLabel,
    FormControl,
    ListItemText,
    OutlinedInput,
    Select,
    SelectChangeEvent,
    MenuItem,
    Checkbox,
} from '@mui/material';
import { MovieConfig, movieConfigNames } from 'shared/types/movies';
import { formatDate } from 'shared/utils/date-format';
import { ConfigurationMovieBox } from './configuration-movie-box';
import { SumbitButton } from 'modules/shared/components/submit-button';
import { Input } from 'modules/shared/components/input';
import { ResetButton } from 'modules/shared/components/reset-button';

interface Props {
    configTitle: 'ADD' | 'EDIT';
    movieConfig: MovieConfig;
    avaliableGenres: string[];
    onSubmitClick: (movieConfig: MovieConfig) => void;
}

const HeightItem = 57;

const configurationStyles = {
    input: {
        width: '100%',
        height: HeightItem,
    },
};

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
    PaperProps: {
        style: {
            maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
        },
    },
};

const requiredFildsName = [
    movieConfigNames.title,
    movieConfigNames.release_date,
    movieConfigNames.poster_path,
    movieConfigNames.vote_average,
    movieConfigNames.genres,
    movieConfigNames.runtime,
    movieConfigNames.overview,
];

const requiredFildFilled = (
    filds: typeof requiredFildsName,
    movieConfig: MovieConfig,
) => {
    return filds.every((field) => !!movieConfig[field as keyof MovieConfig]);
};

export const ConfigurationMovie = ({
    configTitle,
    movieConfig,
    avaliableGenres,
    onSubmitClick,
}: Props) => {
    const [config, setConfig] = React.useState(movieConfig);

    const onKeyChangeHandle = (key: string, value: string | string[]) => {
        setConfig({ ...config, [key]: value });
    };

    const onSubmitHandle = () => {
        onSubmitClick(config);
    };

    const onResetHandle = () => {
        setConfig({
            ...config,
            title: '',
            release_date: '',
            poster_path: '',
            vote_average: 0,
            genres: [],
            runtime: 0,
            overview: '',
        });
    };

    const onHandleGenreChange = (event: SelectChangeEvent<typeof config.genres>) => {
        const {
            target: { value },
        } = event;
        setConfig({
            ...config,
            genres: typeof value === 'string' ? value.split(',') : value,
        });
    };

    const allGenres = [
        ...avaliableGenres,
        ...config.genres.map((g) => g.split(',')),
    ] as string[];
    const submitButtonDisabled = !requiredFildFilled(requiredFildsName, config);

    return (
        <ConfigurationMovieBox>
            <Typography variant='h6' component='h2'>
                {configTitle} MOVIE
            </Typography>
            <br />
            <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
                <Grid item xs={8}>
                    <Input
                        label='Title'
                        variant='outlined'
                        style={configurationStyles.input}
                        value={config.title}
                        onChange={(e) =>
                            onKeyChangeHandle(movieConfigNames.title, e.target.value)
                        }
                    />
                </Grid>
                <Grid item xs={4}>
                    <DatePicker
                        label='Release date'
                        value={config.release_date}
                        onChange={(newValue) => {
                            onKeyChangeHandle(
                                movieConfigNames.release_date,
                                formatDate(newValue),
                            );
                        }}
                        renderInput={(params) => (
                            <Input style={configurationStyles.input} {...params} />
                        )}
                    />
                </Grid>
                <Grid item xs={8}>
                    <Input
                        style={configurationStyles.input}
                        label='Movie Url'
                        variant='outlined'
                        type='url'
                        value={config.poster_path}
                        onChange={(e) =>
                            onKeyChangeHandle(
                                movieConfigNames.poster_path,
                                e.target.value,
                            )
                        }
                    />
                </Grid>
                <Grid item xs={4}>
                    <Input
                        style={configurationStyles.input}
                        variant='outlined'
                        type='number'
                        label='Rating'
                        value={config.vote_average}
                        onChange={(e) =>
                            onKeyChangeHandle(
                                movieConfigNames.vote_average,
                                e.target.value,
                            )
                        }
                    />
                </Grid>
                <Grid item xs={8}>
                    <FormControl sx={{ width: '100%' }}>
                        <InputLabel id='genres-checkbox-label'>Genres</InputLabel>
                        <Select
                            labelId='genres-checkbox-label'
                            multiple
                            value={config.genres}
                            onChange={onHandleGenreChange}
                            input={<OutlinedInput label='Genres' />}
                            renderValue={(selected) => selected.join(', ')}
                            MenuProps={MenuProps}
                        >
                            {allGenres.map((genre, id) => (
                                <MenuItem key={id} value={genre}>
                                    <Checkbox
                                        checked={config.genres.indexOf(genre) > -1}
                                    />
                                    <ListItemText primary={genre} />
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                </Grid>
                <Grid item xs={4}>
                    <Input
                        style={configurationStyles.input}
                        variant='outlined'
                        type='number'
                        label='Runtime'
                        value={config.runtime}
                        onChange={(e) =>
                            onKeyChangeHandle(movieConfigNames.runtime, e.target.value)
                        }
                    />
                </Grid>
                <Grid item xs={12}>
                    <Input
                        multiline
                        style={{ ...configurationStyles.input, height: '100%' }}
                        minRows={3}
                        maxRows={6}
                        label='Overview'
                        value={config.overview}
                        onChange={(e) =>
                            onKeyChangeHandle(movieConfigNames.overview, e.target.value)
                        }
                    />
                </Grid>
                <Grid item xs={6} />

                <Grid item xs={3}>
                    <ResetButton variant='outlined' onClick={onResetHandle}>
                        RESET
                    </ResetButton>
                </Grid>
                <Grid item xs={3}>
                    <SumbitButton
                        variant='contained'
                        endIcon={<SendIcon />}
                        onClick={onSubmitHandle}
                        disabled={submitButtonDisabled}
                    >
                        SUBMIT
                    </SumbitButton>
                </Grid>
            </Grid>
        </ConfigurationMovieBox>
    );
};
