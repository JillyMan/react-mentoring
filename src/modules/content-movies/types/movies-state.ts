import { MovieConfig } from 'shared/types/movies';

export interface MoviesSearchSettings {
    offset: number;
    limit: number;

    sortBy?: string;
    sortOrder?: 'desc' | 'asc';
    searchValue?: string;
    searchBy?: 'title' | 'genres';
    genresFilter: string[];
}

export interface MoviesState {
    movies: MovieConfig[] | null;
    totalAmount: number;

    searchSettings: MoviesSearchSettings;
}
