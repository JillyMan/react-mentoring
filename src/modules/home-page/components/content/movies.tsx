import React from 'react';
import { MovieConfig } from "../../../../shared/types/movies";
import { Grid } from '@mui/material';
import { MovieCardWithError } from './movie-card';

interface Props {
    movies: MovieConfig[];
}

export const Movies = ({ movies }: Props) => {
    return (
        <Grid container rowSpacing={3} spacing={3}>
            {movies.map((value, id) => (
                <Grid item xs={3} key={id}>
                    <MovieCardWithError movie={value} />
                </Grid>
            ))}
        </Grid>
    );
}