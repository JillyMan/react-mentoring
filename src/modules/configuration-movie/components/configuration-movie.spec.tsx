import React from 'react';
import DateAdapter from '@mui/lab/AdapterMoment';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import { render, screen, fireEvent } from '@testing-library/react';
import { ConfigurationMovie } from './configuration-movie';
import { initialMovieConfig, MovieConfig } from 'shared/types/movies';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';

function renderComponent(
    initConfig: MovieConfig,
    onSubmit: (movie: MovieConfig) => void,
) {
    return render(
        <LocalizationProvider dateAdapter={DateAdapter}>
            <ConfigurationMovie
                configTitle='ADD'
                movieConfig={initConfig}
                avaliableGenres={[]}
                onSubmitClick={onSubmit}
            />
        </LocalizationProvider>,
    );
}

describe('Configuration movie', () => {
    test('Sumbit button is not avalible', () => {
        const submitClick = jest.fn();
        renderComponent(initialMovieConfig, submitClick);
        userEvent.click(screen.getByText('SUBMIT'));
        expect(submitClick).not.toHaveBeenCalled();
    });

    it('click reset button to clean current movie config', () => {
        act(() => {
            renderComponent({ ...initialMovieConfig }, jest.fn);

            const title = 'Elden ring';
            const element = screen.getByTestId('movie-title-id');
            fireEvent.change(element, {
                target: { value: title },
            });

            const resetBtn = screen.getByText('RESET');
            resetBtn.click();
        });

        const element = screen.getByTestId('movie-title-id');
        expect(element.value).toBe('');
    });

    test('configuration movie snapshot test', () => {
        const { asFragment } = renderComponent(
            {
                ...initialMovieConfig,
                title: 'new title',
                overview: 'bla bla bla overview',
                vote_average: 10,
                vote_count: 10,
                poster_path: 'https://some-url.com',
                genres: ['one', 'second'],
                runtime: 200,
            },
            jest.fn,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
