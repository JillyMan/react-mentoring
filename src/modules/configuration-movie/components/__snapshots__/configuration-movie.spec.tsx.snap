// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration movie configuration movie snapshot test 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiBox-root css-12bhh9a"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        ADD MOVIE
      </h2>
      <br />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-10mzr36-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-12dpnmv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 57px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Title
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="movie-title-id"
                name="title"
                type="text"
                value="new title"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 57px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Release date
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="true"
                aria-label="Choose date"
                aria-readonly="true"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    Release date
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-12dpnmv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 57px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Movie Url
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                name="poster_path"
                type="url"
                value="https://some-url.com"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Movie Url
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 57px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Rating
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                name="vote_average"
                type="number"
                value="10"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Rating
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-12dpnmv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="genres-checkbox-label"
            >
              Genres
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="genres-checkbox-label mui-component-select-genres"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-genres"
                role="button"
                tabindex="0"
              >
                one, second
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="genres"
                tabindex="-1"
                value="one,second"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Genres
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 57px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Runtime
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                name="runtime"
                type="number"
                value="200"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Runtime
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-11 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="width: 100%; height: 100%;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Overview
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                name="overview"
                style="height: 0px; overflow: hidden;"
              >
                bla bla bla overview
              </textarea>
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-186xcr5"
                >
                  <span>
                    Overview
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-ayy8wd-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-j23p4t-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root ForwardRef(Button)-root-12 ForwardRef(Button)-root-13 css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            RESET
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-j23p4t-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root ForwardRef(Button)-root-14 ForwardRef(Button)-root-15 css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            SUBMIT
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SendIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
